services/dao-service.ts

```diff
@@ -257,13 +302,19 @@ export default class DAO {
   async addAllowedTokens(addresses: string[], isTransactional: boolean) {
     if (!this.registry) await this.loadRegistry();
 
-    return this.registry.addAllowedTokens(addresses, isTransactional);
+    this.web3Connection.options.customTransactionHandler = this.transactionHandler.bind(this) 
+
+    return this.registry.addAllowedTokens(addresses, isTransactional)
+    .finally(() => this.web3Connection.options.customTransactionHandler = null)
```

Comment:
not matching code-style-spacing, this should be alligned with either "registry" or "addAllowedTokens"